src/views/connectionUI.ts

```diff
@@ -335,8 +337,8 @@ export class ConnectionUI {
 		});
 	}
 
-	private handleSelectedConnection(selection: IConnectionCredentialsQuickPickItem): Promise<IConnectionInfo> {
-		return new Promise<IConnectionInfo>((resolve, reject) => {
+	private async handleSelectedConnection(selection: IConnectionCredentialsQuickPickItem): Promise<IConnectionInfo> {
+		return new Promise<IConnectionInfo>(async (resolve, reject) => {
```

Comment:
I can remove the second async from lambda function.

```suggestion
	private async handleSelectedConnection(selection: IConnectionCredentialsQuickPickItem): Promise<IConnectionInfo> {
		return new Promise<IConnectionInfo>((resolve, reject) => {
```

Reply:
Async is just syntactic sugar - all it does is allow you to use the await keyword and automatically wrap return values in promises. So if you don't use await then you can remove async since it doesn't actually do anything in that case. 

All async functions must return a promise, but not all functions that return a promise must be async.

Reply:
It doesn't matter, since there's no awaited call here, will remove it. ☮️